core/src/test/java/org/apache/iceberg/view/TestViewMetadata.java

```diff
@@ -334,6 +334,9 @@ public void viewMetadataAndMetadataChanges() {
             .addVersion(viewVersionOne)
             .addVersion(viewVersionTwo)
             .addVersion(viewVersionThree)
+            .setCurrentVersionId(1)
+            .setCurrentVersionId(2)
+            // this will only keep the last set version id and ignore the previously set ones
```

Comment:
this test already verifies that there's only a single `SetCurrentViewVersion` in `changes`

Reply:
I don't think we need this if we allow duplicates.